smalldata_tools/SmallDataAna_psana.py

```diff
@@ -2426,28 +2428,14 @@ def _broadcast_xtc_dets(self, cubeName):
         
     @staticmethod
     def make_det_data_dset(fout, detname, det_shape, nbins):
+        """ Is that really necessary? """
         # data dset
         dset_name = '{}_data'.format(detname)
         shape = tuple(np.r_[nbins,det_shape])
-        dset = fout.create_dataset(dset_name, shape)
+        dset = fout.create_dataset(dset_name, shape, dtype=float)
```

Comment:
FMI: Is this necessary? Do you want to force float if we had requests by users to make sure their data is stored as the detector native unit16 (however rare that is) or does it cause issues when there is damage in an event and the type is set inconsistenty? Do we want floats or doubles?

Reply:
It's probably fine with undefined type. True that for the rare cases we are using int this would be better.